src/main/ballerina/ballerinax/awslambda/code.bal

```diff
@@ -0,0 +1,98 @@
+import ballerina/io;
+import ballerina/system;
+import ballerina/http;
+import ballerina/time;
+
+public type Context object {
+
+    public string requestId;
+    public int deadlineMs;
+    public string invokedFunctionArn;
+    public string traceId;
+
+    function __init(string requestId, int deadlineMs, string invokedFunctionArn, string traceId) {
+        self.requestId = requestId;
+        self.deadlineMs = deadlineMs;
+        self.invokedFunctionArn = invokedFunctionArn;
+        self.traceId = traceId;
+    }
+
+    function remainingExecutionTime() returns int {
+        int result = time:currentTime().time - self.deadlineMs;
+        if (result < 0) {
+            result = 0;
+        }
+        return result;
+    }
+
+};
+
+map<(function (Context, json) returns (json|error))> functions = { };
```

Comment:
Since the return type is a union, we don't need the enclosing parenthesis.

```suggestion
map<(function (Context, json) returns json|error)> functions = { };
```